### DISCUSSION TITLE
Self integration

### DISCUSSION THREAD

**Discussion by donhello** [searchspring/snap#845] [Q&A] [answered]
Hi,

This is my first time integrating search spring,

I followed this guide (Self-snap)

https://searchspring.github.io/snap/#/start-setup

I'm expecting to see the catalog and working search bar like on the sample project,

![image](https://github.com/searchspring/snap/assets/136287723/2f13e91a-4162-473d-b20d-a9d2721bd412)

but sadly I don't see these and I get this error

![image](https://github.com/searchspring/snap/assets/136287723/75589ad6-216a-43cd-94e8-0fb98eed93f2)

Steps to replicate:
1. Build the snap project on my local 
![image](https://github.com/searchspring/snap/assets/136287723/84fccf4e-edf1-4b1d-b4a7-04229d2afe4e)
![image](https://github.com/searchspring/snap/assets/136287723/edb59175-d9e0-45b6-a45f-04a83ce9b518)

2. upload bundle.js file to our main project
3. referenced the bundle.js file on the main project
4. inserted a div with `id=searchspring-content`
5. Deployed to a draft theme (Shopify)

Note: Did all these on a simple html project on my local, but gets the same error.

**Comment by korgon** [comment, marked as answer]
Hello donhello,

The error is saying that Snap could not find the integration script. I think there may be one piece of information missing from our instructions there - and that is, that the script element for the `bundle.js` file (step 3 in reproduction steps) must have an `id` on it that starts with `searchspring`.

For example, in most cases we use `searchspring-context`:
```
<script id="searchspring-context" src="bundle.js"></script>
```

Hope that helps, if not please let us know!

**Comment by donhello** [comment]
it worked, got new errors saying that I need to include the other bundle files, so I did.

thank for the help @korgon

**Reply by korgon** [reply]
Awesome, sounds like things are working now! Sorry for the trouble - we will work to get that documentation cleared up.

Cheers!
